starlette/datastructures.py

```diff
@@ -438,7 +438,7 @@ async def write(self, data: bytes) -> None:
         if self.size is not None:
             self.size += len(data)
 
-        if self._in_memory:
+        if self._in_memory and self.file.tell() + len(data) <= self.file._max_size:
```

Comment:
The `SpooledTemporaryFile` `rollover` is blocking

Reply:
Should come soon: https://github.com/agronholm/anyio/pull/867